File: Baekjoon/Class2/22.10814 나이순 정렬.java
import java.io.*;
import java.util.Arrays;
import java.util.StringTokenizer;

class Main {
    private static class Person {
        private int age;
        private String name;
        public Person(int age, String name) {
            this.age = age;
            this.name = name;
        }
    }

    public static void main(String[] args) throws IOException {
        int n;
        Person[] people;
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        StringTokenizer st;

        n = Integer.parseInt(br.readLine());
        people = new Person[n];
        for(int i = 0; i < n; i++) {
            st = new StringTokenizer(br.readLine());
            people[i] = new Person(Integer.parseInt(st.nextToken()), st.nextToken());
        }

        Arrays.sort(people, (p1, p2) -> p1.age - p2.age);
        for(Person person: people)
            sb.append(person.age).append(' ').append(person.name).append('\n');
        System.out.print(sb.toString());
        br.close();
    }
}

File: Baekjoon/Class2/22.10814 나이순 정렬.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

struct Person {
    int age;
    string name;
};

bool asc_age(Person p1, Person p2) {
    return p1.age < p2.age;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int n;
    cin >> n;
    vector<Person> people(n);
    for (int i = 0; i < n; i++)
        cin >> people[i].age >> people[i].name;

    stable_sort(people.begin(), people.end(), asc_age);
    for (Person person : people)
        cout << person.age << ' ' << person.name << '\n';
    return 0;
}